packages/libs/error-stack/src/fmt/mod.rs

```diff
@@ -443,8 +499,118 @@ impl From<Style> for OwOStyle {
 #[derive(Debug, Copy, Clone)]
 enum Position {
     First,
-    Middle,
-    Last,
+    Inner,
+    Final,
+}
+
+#[derive(Debug, Copy, Clone)]
+enum Spacing {
+    // Standard to create a width of 4 characters
+    Full,
+    // Minimal width to create a width of 2/3 characters
+    Minimal,
+}
+
+#[derive(Debug, Copy, Clone)]
+struct Indent {
+    /// Is this used in a group context, if that is the case, then add a single leading space
+    group: bool,
+    /// Should the indent be visible, if that isn't the case it will render a space instead of
+    /// `|`
+    visible: bool,
+    /// Should spacing included, this is the difference between `|   ` and `|`
+    spacing: Option<Spacing>,
+}
+
+impl Indent {
+    fn new(group: bool) -> Self {
+        Self {
+            group,
+            visible: true,
+            spacing: Some(Spacing::Full),
+        }
+    }
+
+    fn spacing(mut self, spacing: Spacing) -> Self {
+        self.spacing = Some(spacing);
+
+        self
+    }
+
+    fn no_spacing(mut self) -> Self {
+        self.spacing = None;
+
+        self
+    }
+
+    fn invisible(self) -> Self {
+        self.with_visible(false)
+    }
+
+    fn with_visible(mut self, visible: bool) -> Self {
+        self.visible = visible;
+
+        self
+    }
```

Comment:
I'd only use one variant here:

```suggestion
    fn visible(mut self, visible: bool) -> Self {
        self.visible = visible;
        self
    }
```